cmd/kwil-cli/cmds/common/authinfo.go

```diff
@@ -77,6 +78,26 @@ func convertToHttpCookie(c cookie) *http.Cookie {
 	}
 }
 
+// getDomain returns the domain of the URL.
+func getDomain(target string) (string, error) {
+	if target == "" {
+		return "", fmt.Errorf("target is empty")
+	}
+
+	parsedTarget, err := url.Parse(target)
```

Comment:
One quirk of `url.Parse` is how it handles a string like `localhost:8080`.  It does not error.  It returns a URL with no host and "Opaque":

`&url.URL{Scheme:"localhost", Opaque:"8080", User:(*url.Userinfo)(nil), Host:"", Path:"", RawPath:"", OmitHost:false, ForceQuery:false, RawQuery:"", Fragment:"", RawFragment:""}`

We'll probably be rejecting or erroring on such a target _before_ we reach `getDomain`, but it's still probably an unintended result for `getDomain`:
- input `localhost:8080`
- output `localhost://`

Reply:
ah I got screwed every time I use `url.Parse` 🤦‍♂️